teos/internal_api.py

```diff
@@ -158,6 +158,35 @@ def get_appointment(self, request, context):
 
             return GetAppointmentResponse()
 
+    def get_subscription_info(self, request, context):
+        """Returns a user's subscription information stored in the tower, if the user is registered."""
+        with self.rw_lock.gen_rlock():
+            try:
+                subscription_info = self.watcher.get_subscription_info(request.signature)
+
+                user_struct = Struct()
+
+                user_struct.update(
+                    {
+                        "subscription_expiry": subscription_info.subscription_expiry,
+                        "available_slots": subscription_info.available_slots,
+                        "appointments": list(subscription_info.appointments.keys()),
```

Comment:
Watch out here. `appointments.keys()` are the appointment `uuids`, not the locators. Those ids are internal and unknown by the user. You'll need to pull the locators from the subscription info here.